examples/src/main/kotlin/io/ethers/examples/batchrequests/SimulateBlockTransactions.kt

```diff
@@ -0,0 +1,73 @@
+package io.ethers.examples.batchrequests
+
+import io.ethers.core.isFailure
+import io.ethers.core.types.StateOverride
+import io.ethers.core.types.toBlockOverride
+import io.ethers.core.types.tracers.CallTracer
+import io.ethers.core.types.tracers.MuxTracer
+import io.ethers.core.types.tracers.PrestateTracer
+import io.ethers.core.types.tracers.TracerConfig
+import io.ethers.providers.Provider
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.required
+
+class SimulateBlockTransactions(
+    rpcUrl: String,
+) {
+    // Init provider
+    private val provider = Provider.fromUrl(rpcUrl).unwrap()
+
+    fun run() {
+        // For each new block, simulate all transactions in the block with traceCall
+        provider.subscribeNewHeads().sendAwait().unwrap().forEach { head ->
+            println("New block: ${head.number}")
+
+            val blockWithTransactions = provider.getBlockWithTransactions(head.number).sendAwait().unwrap()
+            val stateOverrides = StateOverride.copy(emptyMap())
+            val blockOverrides = blockWithTransactions.toBlockOverride()
+            for (tx in blockWithTransactions.transactions) {
+                val txReceipt = provider.getTransactionReceipt(tx.hash).sendAwait().unwrap().get()
```

Comment:
I suggest we move this variable next to `provider.traceCall`, and send both async to speed up the execution

Reply:
Understood. Could we potentially send provider.getTransactionReceipt asynchronously while executing provider.traceCall synchronously, since we need those results immediately?

Reply:
Yes, that would work as well